pkg/storegateway/bucket_stores.go

```diff
@@ -71,6 +73,19 @@ type BucketStores struct {
 	tenantsSynced     prometheus.Gauge
 }
 
+type ChunkLimiter struct {
+	limiter *store.Limiter
+}
+
+func (c *ChunkLimiter) Reserve(num uint64) error {
+	err := c.limiter.Reserve(num)
+	if err != nil {
+		return httpgrpc.Errorf(http.StatusUnprocessableEntity, err.Error())
+	}
+
+	return nil
+}
```

Comment:
A couple of nits:
1. I would move it at the end of the file, to keep `NewBucketStores()` close to its `struct` definition
2. I think it's not required to export `ChunkLimiter` so I would rename it to `chunkLimiter`

Reply:
@alanprot could you address this comment, please?

Reply:
Done! :D Thanks again!! :D